### PR TITLE
Fixed #31542 -- Simplified admin JavaScript with "for … of" statements.

### PR DESCRIPTION
**Pull request by jdufresne**
https://code.djangoproject.com/ticket/31542

### CODE REVIEW

**Comment by jdufresne** [comment]
buildbot, test on selenium.

**Comment by jdufresne** [comment]
buildbot, test on selenium.

**Comment by felixxm** [comment]
@jdufresne Thanks :+1: I pushed similar changes to the `SelectBox.js`. Can you take a look?

**Comment by felixxm** [comment]
buildbot, test on selenium.

File: django/contrib/admin/static/admin/js/SelectBox.js [hunk 1]
```diff
@@ -51,8 +42,7 @@
         delete_from_cache: function(id, value) {
             let delete_index = null;
             const cache = SelectBox.cache[id];
-            for (let i = 0, j = cache.length; i < j; i++) {
-                const node = cache[i];
+            for (const [i, node] of cache.entries()) {
```
Comment (pope1ni) [inline django/contrib/admin/static/admin/js/SelectBox.js:45]: :heart: that this syntax is possible.

**Comment by jdufresne** [comment]
Edits look good to me :tada:. Thanks @felixxm

**Comment by felixxm** [comment]
buildbot, test on selenium.

**Pull request merged**